radicle-cli/src/commands/inspect.rs

```diff
@@ -122,82 +112,94 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+        Target::Refs => {
+            let path = storage.path_of(&id).join("refs").join("namespaces");
+
+            Command::new("tree")
+                .current_dir(path)
+                .args(["--noreport", "--prune"])
+                .stdout(Stdio::inherit())
+                .stderr(Stdio::inherit())
+                .spawn()?
+                .wait()?;
+        }
+        Target::Payload => {
             println!(
-                "{} {}",
-                term::format::yellow("commit"),
-                term::format::yellow(oid),
+                "{}",
+                colorizer().colorize_json_str(&serde_json::to_string_pretty(&project.payload)?)?
             );
-            if let Ok(parent) = tip.parent_id(0) {
-                println!("parent {parent}");
-            }
-            println!("blob   {}", blob.id());
-            println!("date   {time}");
-            println!();
-
-            if let Some(msg) = tip.message() {
-                for line in msg.lines() {
-                    term::indented(term::format::dim(line));
+        }
+        Target::History => {
+            let repo = storage.repository(id)?;
+            let head = Doc::<Untrusted>::head(signer.public_key(), &repo)?;
+            let history = repo.revwalk(head)?;
+
+            for oid in history {
+                let oid = oid?.into();
+                let tip = repo.commit(oid)?;
+                let blob = Doc::<Unverified>::blob_at(oid, &repo)?;
+                let content: serde_json::Value = serde_json::from_slice(blob.content())?;
+                let timezone = if tip.time().sign() == '+' {
+                    #[allow(deprecated)]
+                    FixedOffset::east(tip.time().offset_minutes() * 60)
+                } else {
+                    #[allow(deprecated)]
+                    FixedOffset::west(tip.time().offset_minutes() * 60)
+                };
+                let time = DateTime::<Utc>::from(
+                    std::time::UNIX_EPOCH
+                        + std::time::Duration::from_secs(tip.time().seconds() as u64),
+                )
+                .with_timezone(&timezone)
+                .to_rfc2822();
+
+                println!(
+                    "{} {}",
+                    term::format::yellow("commit"),
+                    term::format::yellow(oid),
+                );
+                if let Ok(parent) = tip.parent_id(0) {
+                    println!("parent {parent}");
+                }
+                println!("blob   {}", blob.id());
+                println!("date   {time}");
+                println!();
+
+                if let Some(msg) = tip.message() {
+                    for line in msg.lines() {
+                        if line.is_empty() {
+                            println!();
+                        } else {
+                            term::indented(term::format::dim(line));
+                        }
+                    }
+                    term::blank();
```

Comment:
This is a lot.  Would moving it into its own function help readers?

Reply:
I didn't want to do any unnecessary refactoring here, just left it as-is.